biotrainer/embedders/embedding_service.py

```diff
@@ -68,45 +75,146 @@ def compute_embeddings(self, sequence_file: str, output_dir: Path, protocol: Pro
+        max_embedding_fit = self._max_embedding_fit(embeddings[sequence_ids[0]])
+        
+        if embeddings[sequence_ids[0]] is None:
+            logger.info(f"No embeddings found.")
+            return str(embeddings_file_path)
+        
+        logger.info(f"Embedding dimension: {embeddings[sequence_ids[idx]].shape[-1]}")
+        logger.info("If your dataset contains long reads, it may take more time to process the first few sequences.")
+        
+        with tqdm(total=total_sequences, initial=1, desc="Processing Sequences") as pbar:
+            for idx in range(1, total_sequences):
+                if max_embedding_fit <= 3 or len(embeddings) % max_embedding_fit == 0 or idx == total_sequences - 1:
+                    last_save_id, embeddings = self._save_and_reset_embeddings(embeddings, last_save_id, 
+                                                                               embeddings_file_path, use_reduced_embeddings)
+                    logger.debug(f"New {max_embedding_fit=}")
+                    
+                    embeddings[sequence_ids[idx]] = next(embedding_iter, None)
+                    pbar.update(1)
+                    
+                    # Calculate the new max_embedding_fit for the next batch
+                    max_embedding_fit = self._max_embedding_fit(embeddings[sequence_ids[idx]])
+                    
+                else:
+                    embeddings[sequence_ids[idx]] = next(embedding_iter, None)
+                    pbar.update(1)
+                    
+                    if embeddings[sequence_ids[idx]] is None:
+                        logger.debug(f"len(sequence_ids) > len(embedding_iter) or found a None value in the embedding_iter")
+                        del embeddings[sequence_ids[idx]]
+                        return str(embeddings_file_path)
+                      
         # Save remaining embeddings
         if len(embeddings) > 0:
-            if use_reduced_embeddings:
-                embeddings = self._reduce_embeddings(embeddings, self._embedder)
-            _ = self._save_embeddings(save_id=last_save_id, embeddings=embeddings,
-                                      embeddings_file_path=embeddings_file_path)
+            last_save_id, embeddings = self._save_and_reset_embeddings(embeddings, last_save_id, 
+                                                                       embeddings_file_path, use_reduced_embeddings)
 
-        # Delete embeddings and embedding model from memory now, because they will no longer be needed
+        end_time = time.time()
+        logger.info(f"Time elapsed for saving embeddings: {end_time - start_time:.2f}[s]")
+        
         del embeddings
-        del self._embedder
         gc.collect()
 
         return str(embeddings_file_path)
 
+    @staticmethod
+    def _max_embedding_fit(embedding: ndarray) -> int:
+        """
+        Calculates the maximum number of embeddings that can fit in available memory.
```

Comment:
Please add an explanation to the doc string how you found the magic constants in this function (`0.75` and `18`) and how they can be interpreted